### PR TITLE
Select suggested route

### PR DESCRIPTION
**Pull request by ismailirahim**
**Select a suggested route:**
`/api/vessel-reports/:vessel_report_id/select-route/` :POST

Body example:

```
{
  "suggested_route_id": 3
}
```

**Get the selected suggested_rooute:**
`/api/vessel-report/:vessel_report_id/selected-route/ :GET`

### CODE REVIEW

**Review by zkwsk — CHANGES_REQUESTED** [review]
Sorry I forgot to add this to the task on Trello, but could you add the following functionality under when selecting a suggested route?

Please perform a query to get the `vessel_report` associated with the current `suggested_route`. Now write the `eta` from `sugested_routes.eta` to `vessel_reports.eta`.

The purpose of doing this is that we improve the performance by "caching" the last known eta value on the `vessel_report` where it is a little faster to fetch than on `suggested_route`.

**Review by zkwsk — CHANGES_REQUESTED** [review]
Please add Swagger docs for `/api/vessel-reports/:vessel_report_id/select-route/ :POST` under suggested routes

File: src/server/api/controllers/vessel-reports.controller.js [hunk 1]
```diff
@@ -69,9 +69,59 @@ const getVesselReportById = async id => {
...
+      `Cannot find vessel reports for ID ${id}!`,
+      404
+    );
+  } catch (err) {
+    return err.message;
+  }
+};
+const getSelectedSuggestedRoute = async id => {
+  try {
+    const getSelectedRouteId = await knex("vessel_reports")
```
Comment (zkwsk) [inline src/server/api/controllers/vessel-reports.controller.js]: Rename
```suggestion
    const vesselReports = await knex("vessel_reports")
```

File: src/server/api/controllers/vessel-reports.controller.js [hunk 2]
```diff
@@ -69,9 +69,59 @@ const getVesselReportById = async id => {
...
+      .select("*")
+      .where({ id: id });
 
+    if (
+      getSelectedRouteId !== 0 &&
+      getSelectedRouteId[0].selected_route_id !== null
```
Comment (zkwsk) [inline src/server/api/controllers/vessel-reports.controller.js]:
```suggestion
      vesselReports !== 0 &&
      vesselReports[0].selected_route_id !== null
```

**Review by zkwsk — DISMISSED** [review]
⛱

**Review by Swapna-Sahu — DISMISSED** [review]
Looks good to me

**Review by zkwsk — APPROVED** [review]

**Review by zkwsk — APPROVED** [review]

**Pull request merged**